DQMOffline/Trigger/plugins/HTMonitor.cc

```diff
@@ -0,0 +1,301 @@
+
+}
+
+#include "FWCore/Framework/interface/ESHandle.h"
+#include "FWCore/Framework/interface/EventSetup.h"
+#include "DataFormats/TrackerCommon/interface/TrackerTopology.h"
+#include "Geometry/Records/interface/TrackerTopologyRcd.h"
+void HTMonitor::analyze(edm::Event const& iEvent, edm::EventSetup const& iSetup)  {
+
+  // Filter out events if Trigger Filtering is requested
+  if (den_genTriggerEventFlag_->on() && ! den_genTriggerEventFlag_->accept( iEvent, iSetup) ) return;
+
+  edm::Handle<reco::PFMETCollection> metHandle;
+  iEvent.getByToken( metToken_, metHandle );
+  reco::PFMET pfmet = metHandle->front();
+  if ( ! metSelection_( pfmet ) ) return;
+
+  float ht = 0.0; 
+
+  edm::Handle<reco::PFJetCollection> jetHandle; //add a configurable jet collection & jet pt selection
+  iEvent.getByToken( jetToken_, jetHandle );
+  std::vector<reco::PFJet> jets;
+  jets.clear();
+  if ( jetHandle->size() < njets_ ) return;
+  for ( auto const & j : *jetHandle ) {
+    if ( jetSelection_( j ) ) {
+      jets.push_back(j);
+      if ( jetSelection_HT_(j)) ht += j.pt();
+    }
+  }
+  for ( auto const & j : *jetHandle ) {
+    if ( jetSelection_HT_(j)) ht += j.pt();
+  }
+  if ( jets.size() < njets_ ) return;
+
+  float deltaPhi_met_j1 = 10.0;
+  float deltaPhi_j1_j2 = 10.0;
+
+  if (jets.size() >= 1) deltaPhi_met_j1 = fabs( deltaPhi( pfmet.phi(),  jets[0].phi() ));
+  if (jets.size() >= 2) deltaPhi_j1_j2 = fabs( deltaPhi( jets[0].phi(),  jets[1].phi() ));
+  
+  edm::Handle<reco::GsfElectronCollection> eleHandle;
+  iEvent.getByToken( eleToken_, eleHandle );
+  std::vector<reco::GsfElectron> electrons;
+  electrons.clear();
+  if ( eleHandle->size() < nelectrons_ ) return;
+  for ( auto const & e : *eleHandle ) {
+    if ( eleSelection_( e ) ) electrons.push_back(e);
+  }
+  if ( electrons.size() < nelectrons_ ) return;
+  
+  edm::Handle<reco::VertexCollection> vtxHandle;
+  iEvent.getByToken(vtxToken_, vtxHandle);
+
+  reco::Vertex vtx;
+  math::XYZPoint pv(0, 0, 0);
+  for (vector<reco::Vertex>::const_iterator v = vtxHandle->begin(); v != vtxHandle->end(); ++v) {
+    bool isFake =  v->isFake() ;
+    
+    if (!isFake) {
+      pv.SetXYZ(v->x(), v->y(), v->z());
+      vtx = *v;
+      break;
+    }
+  }
+
+  edm::Handle<reco::MuonCollection> muoHandle;
+  iEvent.getByToken( muoToken_, muoHandle );
+  if ( muoHandle->size() < nmuons_ ) return;
+  std::vector<reco::Muon> muons;
+  muons.clear();
+  for ( auto const & m : *muoHandle ) {
+    if ( muoSelection_( m ) && m.isGlobalMuon() && m.isPFMuon() && m.globalTrack()->normalizedChi2() < 10. && m.globalTrack()->hitPattern().numberOfValidMuonHits() > 0 && m.numberOfMatchedStations() > 1 && fabs(m.muonBestTrack()->dxy(pv)) < 0.2 && fabs(m.muonBestTrack()->dz(pv)) < 0.5 && m.innerTrack()->hitPattern().numberOfValidPixelHits() > 0 && m.innerTrack()->hitPattern().trackerLayersWithMeasurement() > 5 )  muons.push_back(m);
```

Comment:
Same comment as METMonitor, @mtosi - For calculating the efficiency of certain triggers, there is a requirement to select one tight muon (while using the singleMuon dataset). Also, its was not possible to define the same selection cuts in the cff files so I put them here instead. Also, I can not use the function "isTightMuon()" directly, so I put in the tight muon selection cuts one by one in the code.